### ISSUE TITLE
can we #log the status bar itself, so we can re-use the bar?

### ISSUE THREAD

**Issue by technogeeky** [jfelchner/ruby-progressbar#111]
I found a way to re-use the progressbar for multiple purposes during one execution run.

However, it is not apparent that this has happened to the console.

For instance, after using the bar twice during this program, when we are done we see:
```
$ bundle exec kiba etl/customers-to-contacts.etl
[PRE]
[POST]] SkipBadCustomers: |======================================= | 14743/14842
```
-------
However, this misses the fact that bar actually went through a stage earlier:
```
$ bundle exec kiba etl/customers-to-contacts.etl
[PRE]
[READ] customers.dbf: |==                                            | 894/14842
```
then
```
[PRE]
[READ] customers.dbf: |==========================                   | 8818/14842
```
then
```
[PRE]
[TRANS] SkipBadCustomers: |=============                            | 4793/14842
```
and finally
```
[PRE]
[POST]] SkipBadCustomers: |======================================= | 14779/14842
```
------------------
What I would like to see instead is:
```
$ bundle exec kiba etl/customers-to-contacts.etl
[PRE]
[READ] customers.dbf: |============================================| 14842/14842
[POST]] SkipBadCustomers: |======================================= | 14779/14842
```

Is this currently possible, or would it require access to protected or private methods?

Thanks!

**Comment by jfelchner** [comment]
@technogeeky can't you just call `puts` after the bar gets finished?  Then run the bar again?

**Comment by technogeeky** [comment]
@jfelchner Ah, you are correct. There is some weird invocation going on with kiba that is causing things to seem out of order...

Thanks for the prompt reply!

**Comment by jfelchner** [comment]
@technogeeky [a new feature](https://github.com/jfelchner/ruby-progressbar/wiki/Custom-Output-Classes) being released in 1.9 might be of some interest to you.
